src/WebApi.OutputCache.V2/CacheOutputAttribute.cs

```diff
@@ -223,6 +231,11 @@ public override void OnActionExecuting(HttpActionContext actionContext)
                 var responseMediaType = actionExecutedContext.Request.Properties[CurrentRequestMediaType] as MediaTypeHeaderValue ?? GetExpectedMediaType(httpConfig, actionExecutedContext.ActionContext);
                 var cachekey = cacheKeyGenerator.MakeCacheKey(actionExecutedContext.ActionContext, responseMediaType, ExcludeQueryStringFromCacheKey);
 
+                if (IsNoCacheHeaderInRequest(actionExecutedContext.ActionContext))
```

Comment:
this shouldn't be here - once a no cache request has been sent by *any* client, it shouldn't invalidate cache for *all* clients